onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1518,6 +1518,10 @@ Serialized model format will default to ONNX unless:
 )pbdoc")
       .def_readwrite("enable_mem_pattern", &PySessionOptions::enable_mem_pattern,
                      R"pbdoc(Enable the memory pattern optimization. Default is true.)pbdoc")
+      .def_readwrite("enable_mem_reuse", &PySessionOptions::enable_mem_reuse,
+                     R"pbdoc(Enable the memory reuse optimization. Default is true.)pbdoc")
+      .def_readwrite("transfer_ownership_intermediate_output_tensors", &PySessionOptions::transfer_ownership_intermediate_output_tensors,
```

Comment:
I'm sorry but I expect this will be an option exposed to the user, currently all of our benchmarks are done with the assumption ownership of the tensors is with ORT, i.e we hold the reference to yield op input tensors and it so happens after yield op is done executing there is no usage downstream and we free these tensors. Lets leave this here and we can get back to this comment in the very end.

---
In reply to: [600912976](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r600912976) [](ancestors = 600912976)